Add competition rules (#8)
ia-toki/apio-2020-web: rules.md

---
layout: single-page-md
title: APIO 2020 Competition Rules
key: rules
---

> These Competition Rules cover General, Problem Scoring, Contestant Scoring, Grading System, and Clarifications in APIO 2019.

Reviewer: ```suggestion
These Competition Rules cover General, Problem Scoring, Contestant Scoring, Grading System, and Clarifications in APIO 2020.
```

Author: :O Thanks for spotting this. Done.

Delegation Leaders have the responsibility of ensuring that all members of their delegation fully understand these rules and abide by them.

> <span style="color:red">**Please do not share the problems after the contest within 2 days since the competition is not held within a same time range.**</span>

Reviewer 2: ```suggestion
<span style="color:red">**Please do not share the problems after the contest within 2 days since the competition is not held within the same time range.**</span>
```

Author: Done.

<br>

### A. General

> * APIO 2020 contains of **1 trial problemset** and **1 competition day**.

Reviewer: ```suggestion
* APIO 2020 consists of **1 trial problemset** and **1 competition day**.
```

Author: Done.

<br>

### C. Contestant Scoring

* For batch and interactive problems:
  * On creative scoring, for each submission, the score of a subtask is the minimum score among all its test cases.
  * A subtask final score is the **maximum of its score** among all submissions.
  * Contestant's score for a problem is the sum of all its subtasks final score.

  For example, if the first submission got 30 points for the first subtask and 0 points for the second subtask, and the second submission got 0 points for the first subtask and 40 points for the second subtask, then the contestant's score for the problem is 70 points.
* For output-only problems:
  * A test case final score is the **maximum of its score** among all submissions.
  * A contestant's score on a problem is the sum of all its test cases final score.
* A contestant's total score is the sum of **all of problems score**.
* Contestants will be ranked by the total score (descending).
* Two contestants with **the same** total score will achieve **the same** rank.
* Submission time **will not affect** contestant rank at all.

<br>

### D. Grading System

> * [TLX](https://tlx.toki.id) will be used as the grading system.

Reviewer: ```suggestion
* Grading system similar to the public site [TLX](https://tlx.toki.id) will be used.
```

Author: Done.

* Source code limit for each submission is **300 KB**.

> * Contestants will get a full feedback for each of their submission.

Reviewer: ```suggestion
* Contestants will get a full feedback for each submission.
```

Author: Done.

> * Contestants will get a full feedback for each submission.

Reviewer 2: ```suggestion
* Contestants will get full feedback for each submission.
```

Author: Done.

  For example, if the first submission got 30 points for the first subtask and 0 points for the second subtask, and the second submission got 0 points for the first subtask and 40 points for the second subtask, then the contestant's score for the problem is 70 points.
* For output-only problems:
  * A test case final score is the **maximum of its score** among all submissions.
  * A contestant's score on a problem is the sum of all its test cases final score.
* A contestant's total score is the sum of **all of problems score**.
* Contestants will be ranked by the total score (descending).
* Two contestants with **the same** total score will achieve **the same** rank.
* Submission time **will not affect** contestant rank at all.

<br>

### D. Grading System

* [TLX](https://tlx.toki.id) will be used as the grading system.
* Source code limit for each submission is **300 KB**.
* For each test case, the grader will output one of:
  * **AC** (Accepted): program solved the problem within the given time and memory limits.
  * **WA** (Wrong Answer): program stopped within given time and memory limits, but produced the wrong output.

>   * **RTE** (Runtime Error): program crashed or exceedeed the memory limit.

Reviewer 2: ```suggestion
  * **RTE** (Runtime Error): program crashed or exceeded the memory limit.
```

Author: Done. Thanks.
